components/Calendar/Calendar.jsx

```diff
@@ -1,10 +1,166 @@
-  </View>
-);
+import { View, FlatList } from 'react-native';
+import { Calendar } from 'react-native-calendars';
+import styled from 'styled-components';
+import firebase from 'firebase';
+import { useDispatch, useSelector } from 'react-redux';
+import CalendarWorkoutCard from './CalendarWorkoutCard';
+import actions from '../../actions';
+
+const days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
+const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July',
+  'August', 'September', 'October', 'November', 'December'];
+
+const DayTitle = styled.Text`
+  font-family: 'Montserrat_500Medium';
+  font-size: 16px;
+  text-align: left;
+  margin-top: 25px;
+  margin-bottom: 15px;
+`;
+
+const NoWorkoutText = styled(DayTitle)`
+  font-size: 24px;
+`;
+
+function formatDate(date) {
+  return `${days[date.getDay()]}, ${months[date.getMonth()]} ${date.getDate() + 1}`;
+}
+
+function setDates(dates) {
+  const marks = {};
+  dates.forEach((date) => {
+    marks[date] = { marked: true, selectedColor: '#cab0ff' };
+  });
+  return marks;
+}
+
+const CalendarView = () => {
+  const startDate = new Date();
+  const stateStart = `${days[startDate.getDay()]}, ${months[startDate.getMonth()]} ${startDate.getDate()}`;
+  const pastWorkoutDates = useSelector((state) => state.dates.dates);
+  const workoutRecords = useSelector((state) => state.records.records);
+
+  const [selectedDate, setselectedDate] = React.useState(stateStart);
+  const [exercises, setExercises] = React.useState([]);
+  const [markedDates, setMarkedDates] = React.useState(setDates(pastWorkoutDates));
+  const [showWorkout, setShowWorkout] = React.useState(false);
+  const firstRun = React.useRef(true);
+
+  const db = firebase.firestore();
+  const currentUser = firebase.auth().currentUser.uid;
+  const userRef = db.collection('users').doc(currentUser);
+
+  const dispatch = useDispatch();
+
+  const filterRecords = (date, records) => records.filter((e) => e.date === date);
+
+  const getDateRecords = async (user, currentDate) => {
+    const recordCheck = filterRecords(currentDate, workoutRecords);
+    if (recordCheck.length > 0) {
+      console.log('Record found in redux.');
+      setShowWorkout(true);
+      setExercises(recordCheck);
+    } else {
+      const recordsRef = user.collection('workoutRecords');
+      const workoutRecordSnapshot = await recordsRef.where('date', '==', currentDate).get();
+
+      if (workoutRecordSnapshot.empty) {
+        setShowWorkout(false);
+        return;
+      }
+      setShowWorkout(true);
+
+      workoutRecordSnapshot.forEach((doc) => {
+        const data = doc.data();
+        const exerciseRecord = data.exercises;
+        const { date } = data;
+
+        const workoutSets = [];
+
+        exerciseRecord.forEach((record) => {
+          const loggableData = {
+            name: record.exerciseName,
+            sets: record.sets,
+            isReps: true,
+            date,
+          };
```

Comment:
Does this support time based exercises?

Reply:
the isReps field is actually redundant bc time vs reps is determined in the flatlist on line 148 so I'll prob delete it